EarlGrey/Action/GREYMultiFingerSwipeAction.m

```diff
@@ -0,0 +1,167 @@
+//
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "GREYMultiFingerSwipeAction.h"
+
+#import "Action/GREYPathGestureUtils.h"
+#import "Additions/NSError+GREYAdditions.h"
+#import "Additions/NSObject+GREYAdditions.h"
+#import "Additions/NSString+GREYAdditions.h"
+#import "Assertion/GREYAssertionDefines.h"
+#import "Assertion/GREYAssertions+Internal.h"
+#import "Common/GREYError.h"
+#import "Event/GREYSyntheticEvents.h"
+#import "Matcher/GREYAllOf.h"
+#import "Matcher/GREYMatcher.h"
+#import "Matcher/GREYMatchers.h"
+#import "Matcher/GREYNot.h"
+
+@implementation GREYMultiFingerSwipeAction {
+  /**
+   *  The direction in which the content must be scrolled.
+   */
+  GREYDirection _direction;
+  /**
+   *  The duration within which the swipe action must be complete.
+   */
+  CFTimeInterval _duration;
+  /**
+   *  Start point for the swipe specified as a percentage 
+   *  of the swipped element's accessibility frame.
+   */
+  CGPoint _startPercents;
+  /**
+   *  Number of parallel swipes
+   */
+  NSUInteger _numberOfFingers;
+}
+
+- (instancetype)initWithDirection:(GREYDirection)direction
```

Comment:
I am noticing a compilation error in this method:
>convenience initializer missing a 'self' call to another initializer.

I was able to fix it by removing this method all together and moving the contents over to:
```
- (instancetype)initWithDirection:(GREYDirection)direction
                         duration:(CFTimeInterval)duration
                  numberOfFingers:(NSUInteger)numberOfFingers
                    startPercents:(CGPoint)startPercents {
  NSAssert(startPercents.x > 0.0f && startPercents.x < 1.0f,
           @"startPercents must be between 0 and 1, exclusively");
  
  NSAssert(numberOfFingers <= 4, @"No more than four parallel fingers are supported");
  
  NSString *name =
      [NSString stringWithFormat:@"Swipe %@ for duration %g", NSStringFromGREYDirection(direction),
                                duration];
  self = [super initWithName:name
                 constraints:grey_allOf(grey_interactable(),
                                        grey_not(grey_systemAlertViewShown()),
                                        grey_kindOfClass([UIView class]),
                                        grey_respondsToSelector(@selector(accessibilityFrame)),
                                        nil)];
  if (self) {
    _direction = direction;
    _duration = duration;
    _numberOfFingers = numberOfFingers;
    _startPercents = startPercents;
  }
  return self;
}
```

and also had to change the other initializer to call to the one above:
```
- (instancetype)initWithDirection:(GREYDirection)direction
                         duration:(CFTimeInterval)duration
                  numberOfFingers:(NSUInteger)numberOfFingers {
  return [self initWithDirection:direction
                        duration:duration
                 numberOfFingers:numberOfFingers
                   startPercents:CGPointMake(0.5, 0.5)];
}
```